src/main/java/com/github/mauricioaniche/ck/metric/MethodInvocationsLocal.java

```diff
@@ -0,0 +1,78 @@
+    //Recursively extract all method invocations starting with the given method
+    //Explored contains all previously explored invocations
+    //Invocations contains all direct method invocations of interest
+    //The algorithm is a depth first search
+    private Set<String> invocations(String invokedMethod, Set<String> explored, HashMap<String, Set<String>> invocations){
+        explored.add(invokedMethod);
+
+        //only explore local method invocations that were not previously explored
+        Set<String> nextInvocations = invocations.get(invokedMethod).stream()
+                .filter(invoked -> !explored.contains(invoked))
+                .collect(Collectors.toSet());
+        for (String nextInvocation : nextInvocations){
+            explored.addAll(invocations(nextInvocation, explored, invocations));
+        }
+        //Stops when all invocations are explored: there are no more invocations to be explored
+        return explored;
+    }
+
+    //Generate an indirect method invocations map
+    //Method contains all methods of interest
+    //Invocations contains all direct method invocations of interest
+    private HashMap<String, Set<String>> invocationsIndirect(Set<CKMethodResult> methods, HashMap<String, Set<String>> methodInvocationsLocal){
+        HashMap<String, Set<String>> methodInvocationsIndirectLocal = new HashMap<>();
+
+        //extract all indirect local invocations for all methods in the current class
+        for (CKMethodResult method : methods){
+            //extract all local invocations for the current method
+            Set<String> localInvocations =  invocations(method.getQualifiedMethodName(), Sets.newHashSet(ArrayUtils.EMPTY_STRING_ARRAY), methodInvocationsLocal);
+            //remove the method itself and all direct invocations
+            localInvocations.remove(method.getQualifiedMethodName());
+            localInvocations.removeAll(methodInvocationsLocal.get(method.getQualifiedMethodName()));
+
+            methodInvocationsIndirectLocal.put(method.getQualifiedMethodName(), localInvocations);
+        }
+        return methodInvocationsIndirectLocal;
+    }
+
+    //Extract all local(inner-class) method invocations and save them in a HashMap
+    private HashMap<String, Set<String>> extractLocalInvocations(Set<CKMethodResult> methods){
+        HashMap<String, Set<String>> methodInvocationsLocal = new HashMap<>();
+
+        Set<String> methodNames = methods.stream().map(CKMethodResult::getQualifiedMethodName).collect(Collectors.toSet());
+        for (CKMethodResult method : methods){
+            Set<String> invokedLocal =  method.getMethodInvocations().stream()
+                    .filter(methodNames::contains)
+                    .collect(Collectors.toSet());
+            methodInvocationsLocal.put(method.getQualifiedMethodName(), invokedLocal);
+        }
+        return methodInvocationsLocal;
+    }
+
+    @Override
+    public void setResult(CKClassResult result) {
```

Comment:
Interesting, I never formally defined it in the API, but it's indeed true: `setResult(CKClassResult)` is only invoked once the entire class is processed, and thus, you can trust that `result.getMethods()` will have all the methods of the class.

Reply:
Took me a while to figure this out, but once I did I was very happy! :)